### PR TITLE
3️⃣ Introduce renderWeb3Hook

### PR DESCRIPTION
**Pull request by rzadp**
- Closes https://github.com/EthWorks/useDapp/issues/25
- Starts https://github.com/EthWorks/useDapp/issues/17

### CODE REVIEW

File: packages/core/src/constants/abis.ts [hunk 1]
```diff
@@ -16,3 +16,6 @@ export const MULTICALL_ABI = new Interface([
...
...1041e816105d4565b82525050565b600061042f826105c2565b61043981856105c6565b93508360208202850161044b856105bc565b60005b84811015610482578383038852610466838351610409565b9250610471826105bc565b60209890980197915060010161044e565b50909695505050505050565b61041e816105df565b60006104a2826105c2565b6104ac81856105c6565b93506104bc8185602086016105fa565b6104c58161062a565b9093019392505050565b60006104da826105c2565b6104e481856105cf565b93506104f48185602086016105fa565b9290920192915050565b600061026382846104cf565b602081016105188284610415565b92915050565b60208101610518828461048e565b6040810161053a828561048e565b81810360208301526103ae8184610424565b60405181810167ffffffffffffffff8111828210171561056b57600080fd5b604052919050565b600067ffffffffffffffff82111561058a57600080fd5b5060209081020190565b600067ffffffffffffffff8211156105ab57600080fd5b506020601f91909101601f19160190565b60200190565b5190565b90815260200190565b919050565b6000610518826105e2565b90565b6001600160a01b031690565b82818337506000910152565b60005b838110156106155781810151838201526020016105fd565b83811115610624576000848401525b50505050565b601f01601f19169056fea265627a7a72305820978cd44d5ce226bebdf172bdf24918753b9e111e3803cb6249d3ca2860b7a47f6c6578706572696d656e74616cf50037'
```
Comment (marekkirejczyk) [inline packages/core/src/constants/abis.ts:21]: It is already in `MultiCall.json`.

Comment (rzadp) [reply packages/core/src/constants/abis.ts:21]: Not in my `MultiCall.json` 😬

File: packages/core/src/constants/abis.ts [hunk 2]
```diff
@@ -16,3 +16,7 @@ export const MULTICALL_ABI = new Interface([
...
```
Comment (marekkirejczyk) [inline packages/core/src/constants/abis.ts:22]: Duplicate with JSON file

File: packages/core/src/constants/MultiCall.json [hunk 3]
```diff
@@ -0,0 +1,141 @@
+[
```
Comment (marekkirejczyk) [inline packages/core/src/constants/MultiCall.json:1]: src/abi
src/constants/abi

File: packages/core/src/testing/renderWeb3Hook.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import { MockWeb3Wrapper } from './mockWeb3Wrapper'
+import { AddressZero } from '@ethersproject/constants'
+import { waitUntil } from './utils'
+import { MULTICALL_BYTECODE } from '../constants'
+import MultiCall from '../constants/MultiCall.json'
+
+const IdentityWrapper = ({ children }: { children: ReactNode }) => <>{children}</>
+
+const mineBlock = async (provider: MockProvider) => {
+  const [acc] = await provider.getWallets()
```
Comment (marekkirejczyk) [inline packages/core/src/testing/renderWeb3Hook.tsx:16]: `[acc]` -> `[wallet]`
```ts
function getAdminWallet(provider) {
  return await provider.getWallets()[9]
}
```

File: packages/core/src/testing/renderWeb3Hook.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  const tx = await acc.sendTransaction({ to: AddressZero, value: 0 })
+  await tx.wait()
+}
+
+export const renderWeb3Hook = async (
+  hook: (props: unknown) => unknown,
+  options?: RenderHookOptions<unknown> | undefined,
+  pollingInterval = 200
```
Comment (marekkirejczyk) [inline packages/core/src/testing/renderWeb3Hook.tsx:24]: Question: Can we force ethers to check for new block?

File: packages/core/src/testing/renderWeb3Hook.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+const mineBlock = async (provider: MockProvider) => {
+  const [acc] = await provider.getWallets()
+  const tx = await acc.sendTransaction({ to: AddressZero, value: 0 })
+  await tx.wait()
+}
+
+export const renderWeb3Hook = async (
+  hook: (props: unknown) => unknown,
+  options?: RenderHookOptions<unknown> | undefined,
```
Comment (marekkirejczyk) [inline packages/core/src/testing/renderWeb3Hook.tsx:23]:
```ts
options: { renderHook: {}, mockProvider: {} )
```

Comment (rzadp) [reply packages/core/src/testing/renderWeb3Hook.tsx:23]: ✅

File: packages/core/src/testing/renderWeb3Hook.tsx (hunk 4 above)
Comment (rzadp) [reply packages/core/src/testing/renderWeb3Hook.tsx]: ✅

File: packages/core/src/constants/abis.ts (hunk 2 above)
Comment (rzadp) [reply packages/core/src/constants/abis.ts:22]: Note: Use JSON file with bytecode

**Review by marekkirejczyk — APPROVED** [review]

**Pull request merged**